### ISSUE TITLE
Error with add_controls parameter in HDFReadData function

### ISSUE THREAD

**Issue by LeoMurphyWM24** [BaPSF/bapsflib#79]
When trying to couple position data from the 6K Compumotor with signal data from a Langmuir probe using the HDFReadData function, I get an error.
Specifically, HDF5ReadData works perfectly with only `file`, `board`, and `channel` specified. But when I use the `add_controls` keyword parameter to specify a control device ("6K Compumotor") and a receptacle (1 in this case) so that I can couple (x, y, z) positions with the probe data, I get an error that I don't understand.

My code:
```
from bapsflib import lapd
from bapsflib._hdf.utils.hdfreaddata import HDFReadData
lapd_file = lapd.File("Users/Leo/lapd-data/March_2022/01_line_valves85V_7400A.hdf5")

print("File controls:", list(lapd_file.controls))                                    # ['6K Compumotor']
print("6K Compumotor configs:", list(lapd_file.controls['6K Compumotor'].configs))   # [1, 2, 3, 4] 

data_1 = HDFReadData(lapd_file, 1, 1)                                                # OK
data_2 = HDFReadData(lapd_file, 1, 1, add_controls=[('6K Compumotor', 1)])           # Error
```

My output:
```
File controls: ['6K Compumotor']
6K Compumotor configs: [1, 2, 3, 4]

/Users/leo/PycharmProjects/lapd-plasma-analysis/venv2/lib/python3.8/site-packages/bapsflib/_hdf/utils/hdfreaddata.py:209: UserWarning: Digitizer not specified so assuming the 'main_digitizer' (SIS crate) defined in the mappings.
  warn(why)
/Users/leo/PycharmProjects/lapd-plasma-analysis/venv2/lib/python3.8/site-packages/bapsflib/_hdf/maps/digitizers/siscrate.py:891: UserWarning: `config_name` not specified, assuming 'Config_1'.
  warn("`config_name` not specified, assuming '"
/Users/leo/PycharmProjects/lapd-plasma-analysis/venv2/lib/python3.8/site-packages/bapsflib/_hdf/maps/digitizers/siscrate.py:917: UserWarning: No `adc` specified, but only one adc used...assuming adc 'SIS 3302'
  warn("No `adc` specified, but only one adc used..."
/Users/leo/PycharmProjects/lapd-plasma-analysis/venv2/lib/python3.8/site-packages/bapsflib/_hdf/utils/hdfreaddata.py:502: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  data = np.empty(shape, dtype=dtype)
Traceback (most recent call last):
  File "/Users/leo/Library/Application Support/JetBrains/PyCharm2021.3/scratches/scratch_bapsflib_issue.py", line 9, in <module>
    data_2 = HDFReadData(lapd_file, 1, 1, add_controls=[('6K Compumotor', 1)])           # error
  File "/Users/leo/PycharmProjects/lapd-plasma-analysis/venv2/lib/python3.8/site-packages/bapsflib/_hdf/utils/hdfreaddata.py", line 455, in __new__
    cdata = HDFReadControls(hdf_file, controls,
  File "/Users/leo/PycharmProjects/lapd-plasma-analysis/venv2/lib/python3.8/site-packages/bapsflib/_hdf/utils/hdfreadcontrols.py", line 265, in __new__
    build_shotnum_dset_relation(shotnum, cdset_dict[cname],
  File "/Users/leo/PycharmProjects/lapd-plasma-analysis/venv2/lib/python3.8/site-packages/bapsflib/_hdf/utils/helpers.py", line 83, in build_shotnum_dset_relation
    index, sni = build_sndr_for_simple_dset(shotnum, dset,
  File "/Users/leo/PycharmProjects/lapd-plasma-analysis/venv2/lib/python3.8/site-packages/bapsflib/_hdf/utils/helpers.py", line 135, in build_sndr_for_simple_dset
    first_sn, last_sn = dset[[-1, 0], shotnumkey]
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "/Users/leo/PycharmProjects/lapd-plasma-analysis/venv2/lib/python3.8/site-packages/h5py/_hl/dataset.py", line 729, in __getitem__
    return self.fields(names, _prior_dtype=new_dtype)[args]
  File "/Users/leo/PycharmProjects/lapd-plasma-analysis/venv2/lib/python3.8/site-packages/h5py/_hl/dataset.py", line 226, in __getitem__
    data = self._dset.__getitem__(args, new_dtype=self.read_dtype)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "/Users/leo/PycharmProjects/lapd-plasma-analysis/venv2/lib/python3.8/site-packages/h5py/_hl/dataset.py", line 783, in __getitem__
    selection = sel.select(self.shape, args, dataset=self)
  File "/Users/leo/PycharmProjects/lapd-plasma-analysis/venv2/lib/python3.8/site-packages/h5py/_hl/selections.py", line 82, in select
    return selector.make_selection(args)
  File "h5py/_selector.pyx", line 272, in h5py._selector.Selector.make_selection
  File "h5py/_selector.pyx", line 201, in h5py._selector.Selector.apply_args
TypeError: Indexing elements must be in increasing order
```
Am I doing something wrong? At the very least, I would be happy with a clearer error message, although it seems like I'm using HDFReadData as it's supposed to be. I'm basing my code on the example at the [HDFReadData class page](https://bapsflib.readthedocs.io/en/latest/api/bapsflib._hdf.utils.hdfreaddata.HDFReadData.html#bapsflib._hdf.utils.hdfreaddata.HDFReadData). It looks like this same error carries through to the higher-level `file.read_data` function.

**Comment by LeoMurphyWM24** [comment]
It seems like this is an issue with how h5py version 3.0 and higher interprets indexing of datasets. The problem was fixed in PR #65 in August 2021, but there has not been a new bapsflib release since then.

**Comment by LeoMurphyWM24** [comment]
​Hello Erik,
That is fantastic. Thank you for working to get h5py 3.0 compatibility out. My code seems to be working well with 1.0.2, but I will test the beta version once it is released.
Sincerely,
Leo
________________________________

**Comment by rocco8773** [comment]
I released a beta version `v2.0.0b1` today that now drops support for `h5py` `v2.x` in favor of supporting `h5py >= 3.0`.  You can install this directly from PyPI using...

```
pip install bapsflib --pre --upgrade
```

Please let me know how it works and if you find any bugs.  All my tests are passing using `h5py >= 3.0`, so I do think we're good on that front.

**Comment by LeoMurphyWM24** [comment]
Hi Erik,
Yes! I just installed bapsflib v2.0.0b1, and it still works perfectly for my project. Thank you for all your help.
My project is still limited in its bapsflib scope, so I will let you know if I find any more bugs later.
Sincerely,
Leo Murphy
________________________________
